scripts/ci/run-app-host-xcodebuild.sh

```
@@ -11,10 +11,53 @@ max_attempts="${CMUX_APP_HOST_XCODEBUILD_ATTEMPTS:-3}"
+# Principled serialization (the actual fix; the retry below is only a backstop).
+# Invariant: a GUI test host owns the Mac's single login session + testmanagerd
+# while it runs. Two hosts on one self-hosted Mac contend for that one session
+# and drop the test-runner channel. Enforce one app-host test at a time PER
+# MACHINE with a machine-local mutex. atomic mkdir is the lock (portable; no
+# util-linux flock dependency on macOS). The lock dir lives on local disk, so
+# different machines use different locks and cross-machine parallelism is kept.
+lock_dir="${CMUX_APP_HOST_TEST_LOCK_DIR:-${TMPDIR:-/tmp}/cmux-app-host-test.lock}"
+lock_wait_seconds="${CMUX_APP_HOST_TEST_LOCK_WAIT_SECONDS:-2400}"
+lock_stale_seconds="${CMUX_APP_HOST_TEST_LOCK_STALE_SECONDS:-2400}"
+lock_held=0
+release_test_lock() { [ "$lock_held" = "1" ] && rm -rf "$lock_dir" 2>/dev/null || true; }
```

Shellcheck (0.11.0)

[info] This function is never invoked. Check usage (or ignored if invoked indirectly).

(SC2329)